tools/uart_hw_test/uart_test_windows.c

```diff
@@ -0,0 +1,313 @@
+	} else {
+		pass++;
+		printf("[BAUD %6d Tx] PASS\n", baud);
+	}
+
+	PurgeComm(handle, PURGE_RXCLEAR);
+}
+
+static void handshake(HANDLE handle, char c)
+{
+	volatile char ch;
+	int ret;
+	do {
+		ch = 0;
+		WriteFile(handle, (void *)&c, 1, NULL, NULL);
+		ret = poll_read(handle, (void *)&ch, 1, HANDSHAKE_INTERVAL);
+		Sleep(300);
+	} while(ch != c);
+}
+
+static void perform_uart_tests(HANDLE handle)
+{
+	int bauds[] = {CBR_115200, CBR_38400, CBR_19200, CBR_9600, CBR_4800};
+	for (int i = 0; i < sizeof(bauds) / sizeof(int); i++) {
+		configure_handle(handle, bauds[i]);
+		handshake(handle, 't');
+		test_uart_tx(handle, bauds[i]);
+		handshake(handle, 'r');
+		test_uart_rx(handle, bauds[i]);
+		handshake(handle, 'x');
+	}
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+int main(int argc, char *argv[])
+{
+	char p[MAX_PORTNO_LENGTH];
+	char portPath[TTYPATH_LEN] = {0, };
+
+	printf("Enter UART port number : ");
+	gets(p);
+	if (strlen(p) < 2) 
+		snprintf(portPath, TTYPATH_LEN, UART_ALT_PATH_STR, p);
+	else 
+		snprintf(portPath, TTYPATH_LEN, UART_PATH_STR, p);
+
+	HANDLE handle;
+	handle  = CreateFile(portPath, GENERIC_READ | GENERIC_WRITE, 0, NULL, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL);
+
+	if (handle == INVALID_HANDLE_VALUE) {
+		printf("%s open failed!!\nPlease enter correct device port\n", portPath);
+		goto dev_open_err;
+	}
+
+	printf("Opened UART port : %s\n", portPath);
+
+	pass = 0;
+	fail = 0;
+
+	printf("######################## UART Test Start #############################\n");
+	printf("######################################################################\n");
+	printf("Test Scenario 0: UART test only\n");
+	printf("######################################################################\n");
+	perform_uart_tests(handle);
+	printf("\n######################################################################\n");
+	printf("Test Scenario 1: UART test with background FS ops\n");
+	printf("######################################################################\n");
+	perform_uart_tests(handle);
+	printf("\n######################################################################\n");
+	printf("Test Scenario 2: UART test with background console logs\n");
+	printf("######################################################################\n");
+	perform_uart_tests(handle);
+	printf("\n######################################################################\n");
+	printf("Test Scenario 3: UART test with background iperf\n");
+	printf("######################################################################\n");
+	perform_uart_tests(handle);
+	printf("######################################################################\n");
+	printf("################## UART Test End [PASS: %d, FAIL: %d] #################\n", pass, fail);
+
+	CloseHandle(handle);
+dev_open_err:
+	return 0;
+}
```

Comment:
4 commits are there, ig you need to merge all the 4 commits into 1

Reply:
done